lib/LongVectorLoweringPass.cpp

```diff
@@ -291,44 +325,201 @@ Value *LongVectorLoweringPass::visit(Value *V) {
     return it->second;
   }
 
-  if (auto *BinOp = dyn_cast<BinaryOperator>(V)) {
-    if (auto *EquivalentType = getEquivalentType(BinOp->getType())) {
-      IRBuilder<> B(BinOp);
+  if (isa<Argument>(V)) {
```

Comment:
`InstructionMap` is looking like a bit of a misnomer now. Maybe `ValueMap` or `RemappedValues`?